storage/statedb/cache_redis.go

```diff
@@ -120,3 +141,7 @@ func (cache *RedisCache) UpdateStats() interface{} {
 func (cache *RedisCache) SaveToFile(filePath string, concurrency int) error {
 	return nil
 }
+
+func (cache *RedisCache) Close() error {
+	return cache.client.Close()
```

Comment:
```suggestion
	cache.UnsubscribeBlock()
	return cache.client.Close()
```
I think `CloseBlockSubscriptionLoop` can be deleted.

Reply:
You can see that `UnsubscribeBlock()` and `Close()` are called in different timing if you see the full code of `(bc *BlockChain) Stop()`.